### PR TITLE
Allow createauxblock RPC with different coinbase addresses

### PR DESCRIPTION
**Pull request by bmhjyu**
AuxpowMiner::createAuxBlock() only caches a single block for a specific height. If we call createauxblock RPC multiple times with different coinbase payout addresses in that case, they will actually all use the first address. This commit uses a map with the coinbase transaction script hash key as the cache so that we can get the correct results.

### CODE REVIEW

File: src/rpc/auxpow_miner.cpp [hunk 1]
```diff
@@ -82,6 +89,7 @@ AuxpowMiner::getCurrentBlock (const CScript& scriptPubKey, uint256& target)
 
         /* Save in our map of constructed blocks.  */
         pblockCur = &newBlock->block;
+        curBlocks[scriptID] = pblockCur;
```
Comment (domob1812) [inline src/rpc/auxpow_miner.cpp]: Nit: Please change that to use `emplace`.

**Comment by domob1812** [comment]
Thanks for your change, that looks good to me!  I've added a tiny nit comment for the code.  Also, there are two more things:

1) Please add testing for your change, e.g. by extending `test/functional/auxpow_mining.py`.
2) This should be merged to the `auxpow` branch rather than `master` (and then to `master` from there).  So please rebase the change and resubmit a PR for that.  If you prefer, I can also take care of that while merging instead.

**Comment by bmhjyu** [comment]
I will rebase and resubmit with changes to address the comments.

**Comment by bmhjyu** [comment]
#301 will be used as the change needs to be merged to `auxpow` branch.

**Pull request closed without merging**